hi-ml-cpath/src/health_cpath/utils/viz_utils.py

```diff
@@ -196,16 +204,34 @@ def plot_heatmap_overlay(
                                          location_bbox=slide_dict[SlideKey.ORIGIN],
                                          scale_factor=slide_dict[SlideKey.SCALE],
                                          should_upscale_coords=should_upscale_coords)
-    cmap = plt.cm.get_cmap("Reds")
+    cmap = plt.cm.get_cmap("Spectral_r")  # _r reverse the color map so that the highest attention is red
 
     tile_xs, tile_ys = sel_coords.T
     rects = [patches.Rectangle(xy, tile_size, tile_size) for xy in zip(tile_xs, tile_ys)]
 
-    pc = collection.PatchCollection(rects, match_original=True, cmap=cmap, alpha=0.5, edgecolor="black")
+    # line width is set to 0 to avoid the black border around the tiles as the tiles are already colored
+    pc = collection.PatchCollection(rects, match_original=True, cmap=cmap, alpha=0.5, linewidth=0)
     pc.set_array(np.array(attentions))
-    pc.set_clim([0, 1])
-    ax.add_collection(pc)
-    plt.colorbar(pc, ax=ax)
+    ax1.add_collection(pc)
```

Comment:
Should we add the `set_clim` back, to get comparable heatmaps?

Reply:
The reason why I removed it is to be able to spot the difference between low abd high attention regions easily.
<img width="95" alt="image" src="https://user-images.githubusercontent.com/37396332/211062299-0e0270f1-f7a9-4288-94d4-363e1064f5cc.png">
vs without 
<img width="101" alt="image" src="https://user-images.githubusercontent.com/37396332/211062359-d1f63664-2513-445a-8736-a69f7717ad1f.png">

Reply:
and in the case of whole slide inference 
<img width="219" alt="image" src="https://user-images.githubusercontent.com/37396332/211062551-d6982325-47ff-4251-9a3c-3a5620cad3e0.png">

Reply:
Attention values are usually very low set clim to 0-1 messes up the coloring and we end up with all tiles of same color